### PR TITLE
[CHARITY-14] Get charity info

### PR DESCRIPTION
**Pull request by paivazov**
Adds get method for `/charities`  and  `/charities/{org_id}` for retrieving info about organisation and makes small refactor.

**References**
 - #14

### CODE REVIEW

File: api_server/charity/routers.py [hunk 1]
```diff
@@ -4,36 +4,85 @@
...
+
+    Args:
+        charity_service: dependency as business logic instance
+
+    Returns:
+        ResponseBaseSchema object with list of CharityOutputSchema object as response data.
+    """
+    return ResponseBaseSchema(status_code=HTTP_200_OK,
+                              data=[
+                                  CharityShortDescriptionSchema.from_orm(charity)
```
Comment (BorodaUA) [inline api_server/charity/routers.py]: Why for /charities we using this "short" schema?

File: api_server/charity/routers.py [hunk 2]
```diff
@@ -4,36 +4,85 @@
...
+@charities_router.get("/", response_model=ResponseBaseSchema, status_code=HTTP_200_OK)
+async def show_charities_list(charity_service: CharityService = Depends()):
+    """
+    Retrieves info about active organisations.
+
+    Args:
+        charity_service: dependency as business logic instance
+
+    Returns:
+        ResponseBaseSchema object with list of CharityOutputSchema object as response data.
```
Comment (BorodaUA) [inline api_server/charity/routers.py]: What schema do we using here?

**Review by BorodaUA — CHANGES_REQUESTED** [review]

**Review by suslovichb — APPROVED** [review]

File: api_server/charity/routers.py (hunk 1 above)
Comment (paivazov) [reply api_server/charity/routers.py]: According to my logic, we retrieve short description of each organisation like `title` and `description`. 
And after we want to know a little bit more of definite organisation, we pass to `/charities/{id}` endpoint and find out info (e.g. phone number, email etc.)

File: api_server/charity/schemas.py [hunk 3]
```diff
@@ -19,12 +19,20 @@ def __new__(self, name, bases, namespaces, **kwargs):
         return super().__new__(self, name, bases, namespaces, **kwargs)
 
 
+class CharityShortDescriptionSchema(BaseModel):
+    title: str = Field(description="Name of organisation")
+    description: str = Field(description="Short description of organisation")
+
```
Comment (suslovichb) [inline api_server/charity/schemas.py]: Could you please add id here?

File: api_server/charity/routers.py (hunk 2 above)
Comment (paivazov) [reply api_server/charity/routers.py]: changed

**Review by suslovichb — CHANGES_REQUESTED** [review]

File: api_server/charity/schemas.py (hunk 3 above)
Comment (paivazov) [reply api_server/charity/schemas.py]: ok, sure, I will

File: api_server/charity/routers.py (hunk 1 above)
Comment (BorodaUA) [reply api_server/charity/routers.py]: Maybe it is a better idea to include in ```/charities``` all public fields? it is quite unusual to "hide" public fields, and for what reason?

Comment (paivazov) [reply api_server/charity/routers.py]: If They're not being used, why do we need to add them?
If we had more than 40 public fields, we should add them all?

Comment (paivazov) [reply api_server/charity/routers.py]: But ok, i will add this fields, if you think it's necessary

Comment (BorodaUA) [reply api_server/charity/routers.py]: Usually there private and public fields, and it is hard to tell how and where they are gonna be used, therefore it is a good idea to put all public fields out there. This what they for, to be public, is it not?

Comment (paivazov) [reply api_server/charity/routers.py]: I think, it's better to ask somebody like Gosha to solve our discussion. And for now, I've added all public fields to response

**Review by BorodaUA — COMMENTED** [review]
Btw, about charity models, maybe such fields must be unique?
```
organisation_email = Column(String(120))
phone_number = Column(String(15))
```
And on that models note, maybe it is a good idea to use constants for such important numbers as columns length? "Magic numbers" usually a bad practice.

**Comment by paivazov** [comment]
> Btw, about charity models, maybe such fields must be unique?
> 
> ```
> organisation_email = Column(String(120))
> phone_number = Column(String(15))
> ```
> 
> And on that models note, maybe it is a good idea to use constants for such important numbers as columns length? "Magic numbers" usually a bad practice.

I think it's redundant

**Comment by BorodaUA** [comment]
> 

Can you elaborate a bit more? What exactly "redundant"?

**Comment by paivazov** [comment]
> > 
> 
> Can you elaborate a bit more? What exactly "redundant"?

It's redundant to put those numbers into constants in this case.

**Comment by BorodaUA** [comment]
> > > 
> > 
> > 
> > Can you elaborate a bit more? What exactly "redundant"?
> 
> It's redundant to put those numbers into constants in this case.

I just leave this here than: https://en.wikipedia.org/wiki/Magic_number_(programming)
What about unique fields in model then?

**Comment by paivazov** [comment]
> > > > 
> > > 
> > > 
> > > Can you elaborate a bit more? What exactly "redundant"?
> > 
> > 
> > It's redundant to put those numbers into constants in this case.
> 
> I just leave this here than: https://en.wikipedia.org/wiki/Magic_number_(programming) What about unique fields in model then?

I've read this topic. In this case, you don't understand that's length of string?

**Comment by paivazov** [comment]
> What about unique fields in model then?

Probably User has two organisations and to manage them, he wanted to has one contact phone or email because It's easier to manage for him

**Comment by BorodaUA** [comment]
> > > > > 
> > > > 
> > > > 
> > > > Can you elaborate a bit more? What exactly "redundant"?
> > > 
> > > 
> > > It's redundant to put those numbers into constants in this case.
> > 
> > 
> > I just leave this here than: https://en.wikipedia.org/wiki/Magic_number_(programming) What about unique fields in model then?
> 
> I've read this topic. In this case, you don't understand that's length of string?

Well if you read this topic, why asking me? Those numbers are self explanatory aren't they? 😁

**Comment by BorodaUA** [comment]
> > What about unique fields in model then?
> 
> Probably User has two organisations and to manage them, he wanted to has one contact phone or email because It's easier to manage for him

So I can add an organization with same number as your organization?

**Comment by paivazov** [comment]
> So I can add an organization with same number as your organization?

If you digging for such trifles, why don't you create tasks for me in which you will describe how I will have to do. Also, we can sync and discuss all the details before I start doing the task.
By the way, `abc.abstractclassmethod` is deprecated since Python 3.3. For now, Python has `classmethod` instead

**Comment by BorodaUA** [comment]
> > So I can add an organization with same number as your organization?
> 
> If you digging for such trifles, why don't you create tasks for me in which you will describe how I will have to do. Also, we can sync and discuss all the details before I start doing the task. By the way, `abc.abstractclassmethod` is deprecated since Python 3.3. For now, Python has `classmethod` instead

Why should I create a task for you? We create tasks for ourselves. I'm just asking you about the code that you wrote, is it normal that many organizations can have same phone number and email? If you think that it is then merge.

**Comment by paivazov** [comment]
> Why should I create a task for you? We create tasks for ourselves. I'm just asking you about the code that you wrote, is it normal that many organizations can have same phone number and email? If you think that it is then merge.

I think It's normal. Just go to prom.ua and check. There is different organisations with same phone numbers and emails. And nobody gives a fog

**Review by BorodaUA — APPROVED** [review]

**Review by suslovichb — APPROVED** [review]

**Pull request merged**